sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -632,8 +632,8 @@ case class FileSourceScanExec(
         }
       }.groupBy { f =>
         BucketingUtils
-          .getBucketId(new Path(f.filePath).getName)
-          .getOrElse(throw QueryExecutionErrors.invalidBucketFile(f.filePath))
+          .getBucketId(f.toPath.getName)
+          .getOrElse(throw QueryExecutionErrors.invalidBucketFile(f.urlEncodedPath))
```

Comment:
I'm wondering if we should still show the hadoop path string in the error message, as the url-encoded one is hard to read.

Reply:
I tend to agree, because users are always providing the path strings as input (I think?).
But that would be a behavior change I think. Since this is not a behavioral bug, I would prefer to leave that change to later.